openhgnn/sampler/test_MAGNN_sampler.py

```diff
@@ -0,0 +1,209 @@
+# TODO: This file should be deprecated, it's just for testing.
+# TODO: TEST!!!
+import sys
+import os
+sys.path.append('/home/ubuntu/openhgnn/')
+os.chdir('/home/ubuntu/openhgnn/')
+
+from openhgnn.models.MAGNN import MAGNN
+from openhgnn.sampler.MAGNN_sampler import MAGNN_sampler, collate_fn
+from openhgnn.sampler.test_config import CONFIG
+import argparse
+import warnings
+import time
+import dgl
+import numpy as np
+import torch as th
+import torch.nn.functional as F
+from torch.utils.data import DataLoader
+from sklearn.metrics import f1_score
+from sklearn.model_selection import train_test_split
+from sklearn.svm import LinearSVC
+
+class EarlyStopping:
```

Comment:
This reminds me of PyTorch Lightning's callbacks.  Maybe in later integrations you can try that.

Reply:
OK. I'll take a try.